New Post: Cimpler -- continous integration, unix style

danielbeardsley opened the pull request:
A post introducing the CI server that iFixit has been using for 6 months.

danielbeardsley committed:
Cimpler: add initial file with header

danielbeardsley committed:
Cimpler: working on the post

Add a nice style for screenshot images

danielbeardsley committed:
Cimpler: add more content to post

And a screenshot of a failing build on github.

danielbeardsley committed:
Make Headings slightly closer to their content

Add 5px above, remove 5px below.

danielbeardsley committed:
Style: improve look of Lists

danielbeardsley committed:
Cimpler, betterify header

danielbeardsley committed:
Cimpler Post: improve some sentences and add links

Also, move most inline links to the footer.

danielbeardsley committed:
Cimpler Post: move important config to top

danielbeardsley committed:
Cimpler Post: alter post meta-data

danielbeardsley committed:
Cimpler Post: move ifixit section to the top

It shows off what cimpler does better than any of the other sections.
Also, add to a few of the shorter sentences.

danielbeardsley committed:
Cimpler Post: linkify ifixit.com and ask for help.

timothyasp commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+Jenkins and friends ended up being unintuitive (at least for our purposes)
+and had far more features than we needed.
+
+I love node.js and needed something simple, single-purpose, and extensible,
```
What about a dash at the end to add emphasis to "you know, unix-style". http://lilt.ilstu.edu/golson/punctuation/dash.html

"I love node.js and needed something simple, single-purpose, and extensivle - you know, unix-style."

danielbeardsley replied:
Fixed.

timothyasp commented:
Reads well and does a good job explaining cimpler - great work!

marczych commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+title: "Cimpler - continuous integration, unix style"
+author: Daniel Beardsley
+author_url: http://github.com/danielbeardsley
+summary: Cimpler is a continuous intergration server in node.js
```
`integration`

"integration server **written** in node.js" ?

danielbeardsley replied:
Fixed... slightly different.

marczych commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+author: Daniel Beardsley
+author_url: http://github.com/danielbeardsley
+summary: Cimpler is a continuous intergration server in node.js
+         that aims to do one this, and do it well, or at least simply.
```
"to do one **thing**"?

danielbeardsley replied:
Fixed

marczych commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+when it receives a [post-receive][hooks]
+notification from github.
+2. The git-build plugin takes builds off the queue,
+does some merging,
```
Should we explain what merging is done and why?

danielbeardsley replied:
Fixed.

marczych commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+## What it does
+
+**Cimpler is mostly just a queue with some plugins**
+with a few useful special-purpose constructs provided for its plugins to utilize.
```
Two `with`s in the same sentence sounds kinda awkward.

xiongchiamiov replied:
Could just change the second one to an "and": Cimpler is mostly just a queue with some plugins and a few useful constructs...

danielbeardsley replied:
Fixed

marczych commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+module.exports = {
+   /**
+    * All plugin can register connect middleware that
+    * will listen on this port.
```
`plugins`

Do we need both `register` and `connect`?

danielbeardsley replied:
Thanks, but connect is being used as a [noun](http://www.senchalabs.org/connect), not a verb, so `connect.js` it is. `plugin` is fixed though.

xiongchiamiov commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+---
+
+Because the world really needed another CI server, we made one.
+Checkout [cimpler on github][repo] and feel free to contribute.
```
[Checkout](http://www.merriam-webster.com/dictionary/checkout) vs. [check out](http://www.merriam-webster.com/dictionary/check+out) - both are valid nouns _and_ verbs, but with slightly different meanings.  Stupid English.

danielbeardsley replied:
Thanks! Fixed.

xiongchiamiov commented on _posts/2013-03-03-cimpler-continuous-integration-server.md:
```diff
@@ -0,0 +1,154 @@
+
+      'github-commit-status': {
+         auth: {
+            // Anything accepted github.authenticate()
```
Anything accepted _in_?

danielbeardsley replied:
Fixed.

xiongchiamiov commented on stylesheets/styles.css:
```diff
@@ -9,11 +9,21 @@ body {
+  margin:0 0 25px;
+}
+
+/* Make litsts a little closer to their preceeding paragraphs */
```
[*lists](http://harkavagrant.com/index.php?id=302)

danielbeardsley replied:
Fixed.

danielbeardsley committed:
Cimpler Post: lots of little speling and grammar fixes

Thanks to @timotheyasp @xiongchiamiov @marczych for pre-reading and catching
all my derpings.
